packages/superset-ui-legacy-preset-chart-deckgl/src/utils/getPointsFromPolygon.ts

```diff
@@ -0,0 +1,34 @@
+type Point = [number, number];
+
+/** Format originally used by the Polygon plugin */
+type CustomPolygonFeature = {
+  polygon: Point[];
+};
+
+/**
+ * Format that is geojson standard
+ * https://geojson.org/geojson-spec.html
+ */
+type GeojsonPolygonFeature = {
+  polygon: {
+    type: 'Feature';
+    geometry: {
+      type: 'Polygon';
+      coordinates: Point[][];
+    };
+  };
+};
+
+function isCustomPolygonFeature(
+  feature: CustomPolygonFeature | GeojsonPolygonFeature,
+): feature is CustomPolygonFeature {
+  return Array.isArray(feature.polygon);
+}
+
+export default function getPointsFromPolygon(
+  feature: CustomPolygonFeature | GeojsonPolygonFeature,
+) {
+  return isCustomPolygonFeature(feature)
+    ? feature.polygon
+    : feature.polygon.geometry.coordinates[0];
+}
```

Comment:
Can also just do 

```js
export default function getPointsFromPolygon(
  feature: CustomPolygonFeature | GeojsonPolygonFeature,
) {
  return feature.polygon.geometry?.coordinates[0] || feature.polygon;
}
```

Reply:
Typescript won't recognize `geometry` and can't use `.geometry?` because there are chance this field does not exist at all (when it is `CustomPolygonFeature`). `?` only handles if the field exist but value is `undefined`.

However, I built on top of your idea and added another ternary check to ensure typescript that `geometry` exist.

Reply:
Oh, I think I meant `return feature.polygon?.geometry.coordinates[0] || feature.polygon;` if that works.

Reply:
Has to be
```ts
return 'geometry' in feature.polygon ? feature.polygon.geometry.coordinates[0] : feature.polygon;
```